js/libs/form.js

```diff
@@ -135,9 +185,15 @@ Fliplet().then(function() {
       if (fields.length && (data.saveProgress && typeof progress === 'object') || entry) {
         fields.forEach(function(field) {
           if (entry && entry.data && field.populateOnUpdate !== false) {
-            var fieldData = entry.data[field.defaultValueKey || field.name];
+            var fieldKey = isResetAction ? field.defaultValueKey || field.name : field.name || field.defaultValueKey;
+
+            var fieldData = entry.data[fieldKey];
+
+            if (typeof fieldData === 'undefined') {
```

Comment:
@NikitaJadhavXoriant Isn't the fallback to `entry.data[field.name] || entry.data[field.defaultValueKey]` considered unnecessary in this case? Since `fieldKey` is assigned the value of either `field.defaultValueKey` or `field.name` based on the condition, it covers the possibility of those two properties being `undefined` or not present.

If I'm wrong, please give me an example of set of values where this is necessary.

Reply:
Hi @tonytlwu, please consider below example for above scenario.
Thanks

https://github.com/Fliplet/fliplet-widget-form-builder/assets/108272606/355d9141-1737-4abc-9ea0-edeff77d1376

cc @yogeshmsharma-xoriant

Reply:
@NikitaJadhavXoriant Thanks for the video. Can you clarify what would be the value of the following variables at [this point](https://github.com/Fliplet/fliplet-widget-form-builder/pull/499/files/919a135b1d984035cbbf103cfa459da09d1074ce#diff-524255b142ab5cdf4d2376cd9ecbc907441103c5db0a458e3bb3e91ce00ac8ffR188) when this happens?

- `isResetAction`
- `field.defaultValueKey`
- `field.name`
- `entry.data`

Reply:
Hi @tonytlwu 

Please find below values 

- `isResetAction`  -  (true if user hit clear button)
- `field.defaultValueKey` - (column name sets by user)
- `field.name` - (field name sets by user)
- `entry.data` - (record from DS)

cc @yogeshmsharma-xoriant